pkg/egressservice/egressservice.go

```diff
@@ -0,0 +1,361 @@
+package egressservice
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	egresssvcv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressservice/v1"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// EgressServiceBuilder provides a struct for EgressService object.
+type EgressServiceBuilder struct {
+	// EgressService definition, used to create the EgressService object.
+	Definition *egresssvcv1.EgressService
+	// Created EgressService object.
+	Object *egresssvcv1.EgressService
+	// Used to store latest error message upon defining or mutating EgressService definition.
+	errorMsg string
+	// api client to interact with the cluster.
+	apiClient goclient.Client
+}
+
+// NewEgressServiceBuilder creates a new instance of EgressService builder.
+func NewEgressServiceBuilder(
+	apiClient *clients.Settings, nsname, name, sourceIPBy string) *EgressServiceBuilder {
+	glog.V(100).Infof(
+		"Initializing new EgressService structure with the following params: "+
+			"name: %s; namespace: %s; sourceIPBy: %s",
+		name, nsname, sourceIPBy)
+
+	if apiClient == nil {
+		glog.V(100).Infof("EgressService 'apiClient' cannot be empty")
+
+		return nil
+	}
+
+	err := apiClient.AttachScheme(egresssvcv1.AddToScheme)
+	if err != nil {
+		glog.V(100).Infof("Failed to add 'egressservice' scheme to client schemes")
+
+		return nil
+	}
+
+	builder := &EgressServiceBuilder{
+		apiClient: apiClient.Client,
+		Definition: &egresssvcv1.EgressService{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+			Spec: egresssvcv1.EgressServiceSpec{
+				SourceIPBy: egresssvcv1.SourceIPMode(sourceIPBy),
+			},
+		},
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the EgressService is empty")
+
+		builder.errorMsg = "The namespace of the EgressService is empty"
+	}
+
+	if name == "" {
```

Comment:
nit:
can we move the name to above the nsname ?

Reply:
sure, updated